tls/s2n_server_finished.c

```diff
@@ -73,12 +69,11 @@ int s2n_server_finished_send(struct s2n_connection *conn)
     POSIX_ENSURE_REF(conn);
 
     uint8_t *verify_data = conn->handshake.server_finished;
-    uint8_t *seq_num = conn->secure->server_sequence_number;
     POSIX_GUARD(s2n_prf_server_finished(conn));
-    POSIX_GUARD_RESULT(s2n_finished_send(conn, seq_num, verify_data));
 
-    POSIX_ENSURE_REF(conn->secure);
-    conn->server = conn->secure;
+    POSIX_GUARD_RESULT(s2n_crypto_parameters_switch(conn));
+
+    POSIX_GUARD_RESULT(s2n_finished_send(conn, verify_data));
```

Comment:
Nit: wouldn't
```
uint8_t *verify_data = conn->handshake.server_finished;
POSIX_GUARD(s2n_prf_server_finished(conn));
POSIX_GUARD_RESULT(s2n_finished_send(conn, verify_data));
POSIX_GUARD_RESULT(s2n_crypto_parameters_switch(conn))
```
be a bit cleaner? Interrupting the finished logic with the crypto parameters switch feels a little odd.